### PR TITLE
first draft of release process

### PR DESCRIPTION
**Pull request by bkmartinjr**
on the assumption that we want a PyPi package, which contains pre-built JS assets, I have started a rough cut at a release process (so we all do it the same way).

Ideally we would use this for the initial release.

@freeman-lab - it is a bit thin on the PyPi publication.  Can you expand?

### CODE REVIEW

File: docs/release_process.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# cellxgene Release Process
```
Comment (csweaver) [inline docs/release_process.md]: Minor nit. With the lowercase cellxgene it looks weird to have capitalized words after it. Can we do "Release Process for cellxgene" instead?

Comment (freeman-lab) [reply docs/release_process.md]: or `#cellxgene release process` :) (that's my personal preference)

Comment (csweaver) [reply docs/release_process.md]: Jeremy and lowercase... I'm fine with it either way

**Comment by freeman-lab** [comment]
key steps to add re: publishing to pypi

```
7. Publish to pypi by performing the following steps (assumes you have `setuptools` and `twine` installed and that you have registered for pypi and have access to write to the cellxgene pypi package) 
- build the distribution by calling `python setup.py sdist bdist_wheel` inside the top directory
- [optional] upload the package to test pypi `twine upload --repository-url https://test.pypi.org/legacy/ dist/*`
- [optional] test the test installation in a fresh virtual environment using `pip install --index-url https://test.pypi.org/simple/ cellxgene`
- upload the package to real pypi using `twine upload dist/*`
- [optional] test the installation in a fresh virtual environment using `pip install cellxgene`
The options steps are for testing purposes, and are recommended for publishing any major releases, and any releases that significantly change the packaging (e.g. new bundled files, new dependencies, etc.)
```

File: docs/release_process.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+_This document defines the release process for cellxgene_
+
+## Overview
+
+The goal of the release process is to publish an installable package
+to PyPi, with a matching tagged release on github.
+
+The release process should result in the following side-effects:
+
+- Version number bump
```
Comment (freeman-lab) [inline docs/release_process.md]: clarify `(using semantic versioning)` (perhaps obvious)

**Comment by freeman-lab** [comment]
added stuff re: pypi, looks good to me otherwise!

File: docs/release_process.md (hunk 1 above)
Comment (bkmartinjr) [reply docs/release_process.md]: resolved

File: docs/release_process.md (hunk 2 above)
Comment (bkmartinjr) [reply docs/release_process.md]: fixed

**Comment by bkmartinjr** [comment]
@freeman-lab - added your PyPi info.

**Comment by freeman-lab** [comment]
looks good to me!

**Pull request merged**